kernel: bump 5.15 to 5.15.102

graysky2 opened the pull request:
No patches rebased as a function of this bump.

graysky2 commented:
@hauke @rsalvaterra - this one is tough, the next patch and the subsequent are over my head.  I also see the one for ipq807x is needing more expert eyes,

robimarko commented:
Great, they already backported the IPQ807x PCIe DTS fixes.
I can prepare a commit for you to just cherry-pick

Borromini commented:
@graysky2 Is `include/kernel-5.15` being left out an oversight or intentional atm?

graysky2 commented:
@Borromini - it is intentionally left out because running ` update_kernel.sh` is in progress.  I am basically stuck at the step of manually rebasing `target/linux/bcm27xx/patches-5.15/950-0596-drm-vc4-dpi-Add-option-for-inverting-pixel-clock-and.patch`

There will be some challenges with subsequent patches as well so this draft PR is basically a request for help showing others the little progress I made thus far.

robimarko commented:
@graysky2 As far as ipq807x goes, here is a branch for 5.15.99
https://github.com/robimarko/openwrt/commits/ipq807x-5.15.99

graysky2 commented:
@robimarko - any chance you can look at bcm27xx .. that is where I am currently stuck

mj22226 commented:
> manually rebasing `target/linux/bcm27xx/patches-5.15/950-0596-drm-vc4-dpi-Add-option-for-inverting-pixel-clock-and.patch`

@graysky2   that patch was merged  https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/gpu/drm/vc4/vc4_dpi.c?h=v5.15.100&id=8e04aaffb6de5f1ae61de7b671c1531172ccf429

I manually rebased  `950-0421-Support-RPi-DPI-interface-in-mode6-for-18-bit-color.patch `https://github.com/mj22226/openwrt/blob/rpi4/target/linux/bcm27xx/patches-5.15/950-0421-Support-RPi-DPI-interface-in-mode6-for-18-bit-color.patch
`950-0707-drm-vc4-For-DPI-MEDIA_BUS_FMT_RGB565_1X16-is-mode-1-.patch `https://github.com/mj22226/openwrt/blob/rpi4/target/linux/bcm27xx/patches-5.15/950-0706-media-i2c-imx219-Scale-the-pixel-clock-rate-for-the-.patch

graysky2 commented:
Ah, 5.15.100 may simplify this.

mj22226 commented:
> Ah, 5.15.100 may simplify this.

@graysky2  bcm2711 is failing to build on 5.15.100  
`ERROR: module 'openwrt/build_dir/target-aarch64_cortex-a72_musl/linux-bcm27xx_bcm2711/linux-5.15.100/net/sched/cls_tcindex.ko' is missing.`  because of this https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/net/sched?h=v5.15.100&id=7c183dc0af472dec33d2c0786a5e356baa8cad19

graysky2 committed:
x86: set deprecated CONFIG_MICROCODE_OLD_INTERACE

We use late loading[1] so need to set this option despite upstream adding a
kernel taint when this option is set.  See discussion in PR#12149 for more details.

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/arch/x86/Kconfig?id=v5.15.100&id2=v5.15.98

1. https://github.com/openwrt/openwrt/blob/master/target/linux/x86/base-files/lib/preinit/02_load_x86_ucode

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 committed:
kernel: tcindex classifier has been retired

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/net/sched?h=v5.15.100&id=7c183dc0af472dec33d2c0786a5e356baa8cad19

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 commented:
@mj22226 - Thank you for point that commit out.  Is it as simple as removing the commented option from `target/linux/generic/config-5.15`

graysky2 commented:
Thank you both @robimarko @mj22226 for your help with this PR :heart:

graysky2 commented:
I should have cherry-picked @mj22226's commit so that he is credited with contributing?  To do this, I am thinking:
* `git reset --hard xxxx`
* add his repo as a remote, fetch and cherry-pick
* run the update script again

Is there another better way?

lovehackintosh commented:
@graysky2 you just need to run
```
wget https://github.com/mj22226/openwrt/commit/3871f03d931e82f1c87b7b915503e7c6dc3b09e7.patch
git am 3871f03d931e82f1c87b7b915503e7c6dc3b09e7.patch
```

graysky2 commented:
@lovehackintosh - I manually added back those patches, then:
```
wget https://github.com/mj22226/openwrt/commit/5672e9690b7aa71a4365769b490b3975fc0a7a25.patch
git am 5672e9690b7aa71a4365769b490b3975fc0a7a25.patch
```
Should I now rebase/fixup the new commit onto my bump 5.15 to 5.15.100?  Will that retain @mj22226's connection to the PR?

hauke commented on target/linux/ramips/patches-5.15/810-uvc-add-iPassion-iP2970-support.patch:
```diff
@@ -225,18 +225,16 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
  #define UVC_QUIRK_FORCE_BPP		0x00001000
--
 +#define UVC_QUIRK_MOTION		0x00001000
 +#define UVC_QUIRK_SINGLE_ISO		0x00002000
```
Both of these defines are overlapping with previous definitions. 
Please do something like this:
````
#define UVC_QUIRK_MOTION		0x00004000
#define UVC_QUIRK_SINGLE_ISO		0x00008000
````

hauke commented on target/linux/bcm27xx/patches-5.15/950-0706-media-i2c-imx219-Scale-the-pixel-clock-rate-for-the-.patch:
```diff
@@ -15,44 +15,44 @@ Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
-+
+ 
 +	/* Relative pixel clock rate factor for the mode. */
 +	unsigned int rate_factor;
```
Please add the new attribute to the end of the struct.

graysky2 replied:
Done, will push shortly.

hauke commented on target/linux/x86/config-5.15:
```diff
@@ -221,7 +221,7 @@ CONFIG_MEMFD_CREATE=y
 CONFIG_MICROCODE_AMD=y
 CONFIG_MICROCODE_INTEL=y
-CONFIG_MICROCODE_OLD_INTERFACE=y
+# CONFIG_MICROCODE_LATE_LOADING is not set
```
Are you sure we are not using this late interface?

I do not know how the microcode loading in OpenWrt works.

graysky2 replied:
I am not sure but the symbol has been removed upstream.  As you suggesting that we set `CONFIG_MICROCODE_LATE_LOADING=y`?  There is a prominent warning in the help against it.

hauke replied:
I do not know. Does someone have an x86 system, with OpenWrt and could check how we install the microcode updates please.

oskarirauta replied:
@hauke 

I have, but I do early loading.
Here's microcode documentation though: https://www.kernel.org/doc/html/next/x86/microcode.html

and a summary..

Early loading:
microcodes must be on cpio namespace (initrd/initramfs) and located flowingly (mine though differed from this..)
 - on Intel: kernel/x86/microcode/GenuineIntel.bin
 - on AMD: kernel/x86/microcode/AuthenticAMD.bin

Late loading:
`echo 1 > /sys/devices/system/cpu/microcode/reload` triggers loading.
> The loading mechanism looks for microcode blobs in /lib/firmware/{intel-ucode,amd-ucode}

And here's some reading about the *change* you mentioned https://www.phoronix.com/news/Linux-5.19-Late-ucode-Loading

hauke replied:
OpenWrt is sort of special and does not use an initramfs, but it could be that the rootfs is already available early enough. I do not know how the boot process on x86 works in detail.

hauke replied:
We use the late loading, see https://github.com/openwrt/openwrt/blob/master/target/linux/x86/base-files/lib/preinit/02_load_x86_ucode
We should activate `CONFIG_MICROCODE_LATE_LOADING` for x86.

oskarirauta replied:
@hauke

rootfs available early or not does not change that any other than cpio is unavailable option for early loading of ucode. I also tested, ucode on rootfs (ext4), does not get loaded at all on any point.

And sure openwrt uses initramfs; or at least supports it, its basic option in kernel and enabled as default, and it should. Openwrt though doesn't provide ucode **as** initramfs- but ucode is necessary **only** for x86_64, so there are variety of boot loaders available for it - and afaik **all** of them support initrd/initramfs.
And even if they wouldn't; just put initrd=intel-ucode.bin to your kernel cmd-line and you probably are all set with just that.

ucode though isn't provided as cpio archive on openwrt, if you meant that; I don't know what should be done for that. On mine, I mount bios boot to /boot and EFI Boot support to /boot/EFI - so ucode goes to /boot... But this isn't _set_ directory structure, so where to extract package containing ucode then? It takes about 6mb if I remember correctly so wouldn't be that interested of storing a backup of it in.. let's say /usr/share/ucode for example. Even though x86_64 systems usually have modular storage options that allow quite large "roms" (hard disks) where 6mb should not be an issue.

ucode initramfs does not need to have init or anything else besides firmware in preset path.

graysky2 replied:
Thanks for the discussion and for the digging you two.  I enabled it and added some verbiage to the commit message doing so.

graysky2 committed:
kernel: bump 5.15 to 5.15.100

Manually rebased:
        bcm27xx/patches-5.15/950-0421-Support-RPi-DPI-interface-in-mode6-for-18-bit-color.patch
        bcm27xx/patches-5.15/950-0706-media-i2c-imx219-Scale-the-pixel-clock-rate-for-the-.patch
        ramips/patches-5.15/810-uvc-add-iPassion-iP2970-support.patch

Removed upstreamed:
        bcm27xx/patches-5.15/950-0707-drm-vc4-For-DPI-MEDIA_BUS_FMT_RGB565_1X16-is-mode-1-.patch[1]
        bcm27xx/patches-5.15/950-0596-drm-vc4-dpi-Add-option-for-inverting-pixel-clock-and.patch[2]
        ipq807x/0006-v5.16-arm64-dts-qcom-Fix-IPQ8074-PCIe-PHY-nodes.patch [3]
        ipq807x/0034-v6.1-arm64-dts-qcom-ipq8074-fix-PCIe-PHY-serdes-size.patch [4]
        ipq807x/0103-arm64-dts-qcom-ipq8074-fix-Gen2-PCIe-QMP-PHY.patch [5]
        ipq807x/0104-arm64-dts-qcom-ipq8074-fix-Gen3-PCIe-QMP-PHY.patch [6]
        ipq807x/0105-arm64-dts-qcom-ipq8074-correct-Gen2-PCIe-ranges.patch [7]
        ipq807x/0108-arm64-dts-qcom-ipq8074-fix-Gen3-PCIe-node.patch [8]
        ipq807x/0109-arm64-dts-qcom-ipq8074-correct-PCIe-QMP-PHY-output-c.patch [9]
        ipq807x/0132-arm64-dts-qcom-ipq8074-correct-USB3-QMP-PHY-s-clock-.patch [10]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.99&id=d2991e6b30020e286f2dd9d3b4f43548c547caa6
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/gpu/drm/vc4/vc4_dpi.c?h=v5.15.100&id=8e04aaffb6de5f1ae61de7b671c1531172ccf429
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=a55a645aa303a3f7ec37db69822d5420657626da
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=d9df682bcea57fa25f37bbf17eae56fa05662635
5. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=7e6eeb5fb3aa9e5feffdb6e137dcc06f5f6410e1
6. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=e88204931d9a60634cd50bbc679f045439c4b91d
7.  https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=1563af0f28afd3b6d64ac79a2aecced3969c90bf
8. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=feb8c71f015d416f1afe90e1f62cf51e47376c67
9. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=69c7a270357a7d50ffd3471b14c60250041200e3
10. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=v5.15.99&id=dd3d021ae5471d98adf81f1e897431c8657d0a18

Build system: x86_64
Build-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3
Run-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>
Tested-by: Robert Marko <robimarko@gmail.com> #ipq807x/Dynalink WRX36
Tested-by: Stefan Lippers-Hollmann <s.l-h@gmx.de> #ipq807x/ax3600, x86_64/FW-7543B, ath79/tl-wdr3600, ipq806x/g10, ipq806x/nbg6817

lovehackintosh commented:
> Should I now rebase/fixup the new commit onto my bump 5.15 to 5.15.100? Will that retain @mj22226's connection to the PR?

Yes, you should. I think that will retain connection.

graysky2 commented:
> Yes, you should. I think that will retain connection.

I am resetting for 5.15.101 now, let's see when I push it.

graysky2 committed:
kernel: bump 5.15 to 5.15.101

No patches rebased as a function of this bump.

Build system: x86_64
Build-tested: ramips/tplink_archer-a6-v3
Run-tested: ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>

Borromini commented:
I don't think 5.15.101 matters for non x86 systems, so my tests so far were for 5.15.100

Compile-tested: mvebu/cortexa9, mvebu/cortexa72, realtek/rtl838x
Run-tested: mvebu/cortexa9 (WRT1200AC), realtek/rtl838x (GS1900-8HP v1)

graysky2 committed:
kernel: bump 5.15 to 5.15.102

No patches needed to be rebased.

Signed-off-by: John Audia <therealgraysky@proton.me>
Tested-by: Robert Marko <robimarko@gmail.com> #ipq807x/Dynalink WRX36
Tested-by: Stefan Lippers-Hollmann <s.l-h@gmx.de> #ipq807x/ax3600, x86_64/FW-7543B, ath79/tl-wdr3600, ipq806x/g10, ipq806x/nbg6817

timocapa commented:
Build and run tested on

Compile-tested: ramips/mt7621
Run-tested: ramips/mt7621 (Xiaomi Mi 3G and Xiaomi Redmi 2100)

![image](https://user-images.githubusercontent.com/22844613/225182502-567d781f-16b5-4bcc-b46a-6e143f2ae938.png)

screenshot powered by nvidia + wayland (missing left window corners)

robimarko commented:
`Tested-by: Robert Marko <robimarko@gmail.com>` #ipq807x/Dynalink WRX36

PerkelSimon commented:
I have problem with linksys e8450/belkin rt3200. The patch applied ok but compilation failed many times.

rmandrad commented:
> I have problem with linksys e8450/belkin rt3200. The patch applied ok but compilation failed many times. If I remove the patch it builds fine if not, here the errors:
> 
> ```
> `make[4]: Entering directory '/run/media/james/D/test_e8450/master/build_dir/target-aarch64_cortex-a53_musl/gnutls-3.7.8'
>  cd . && automake-1.16 --foreign
> configure.ac:41: error: required file 'build-aux/compile' not found
> configure.ac:41:   'automake --add-missing' can install 'compile'
> configure.ac:31: error: required file 'build-aux/missing' not found
> configure.ac:31:   'automake --add-missing' can install 'missing'
> aminclude_static.am:52: warning: if $(CODE_COVERAGE_BRANCH_COVERAGE: non-POSIX variable name
> aminclude_static.am:52: (probably a GNU make extension)
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:59: warning: if $(CODE_COVERAGE_BRANCH_COVERAGE: non-POSIX variable name
> aminclude_static.am:59: (probably a GNU make extension)
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:63: warning: ':='-style assignments are not portable
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:64: warning: code_coverage_v_lcov_cap_$(V: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:65: warning: code_coverage_v_lcov_cap_$(AM_DEFAULT_VERBOSITY: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:67: warning: code_coverage_v_lcov_ign_$(V: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:68: warning: code_coverage_v_lcov_ign_$(AM_DEFAULT_VERBOSITY: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:70: warning: code_coverage_v_genhtml_$(V: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:71: warning: code_coverage_v_genhtml_$(AM_DEFAULT_VERBOSITY: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:73: warning: code_coverage_quiet_$(V: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:74: warning: code_coverage_quiet_$(AM_DEFAULT_VERBOSITY: non-POSIX recursive variable expansion
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:78: warning: subst -,_,$(subst .,_,$(1: non-POSIX variable name
> aminclude_static.am:78: (probably a GNU make extension)
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:87: warning: addprefix --directory ,$(CODE_COVERAGE_DIRECTORY: non-POSIX variable name
> aminclude_static.am:87: (probably a GNU make extension)
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:87: warning: call code_coverage_sanitize,$(PACKAGE_NAME: non-POSIX variable name
> aminclude_static.am:87: (probably a GNU make extension)
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:88: warning: addprefix --directory ,$(CODE_COVERAGE_DIRECTORY: non-POSIX variable name
> aminclude_static.am:88: (probably a GNU make extension)
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:90: warning: addprefix --prefix ,$(CODE_COVERAGE_DIRECTORY: non-POSIX variable name
> aminclude_static.am:90: (probably a GNU make extension)
> Makefile.am:176:   'aminclude_static.am' included from here
> aminclude_static.am:100: warning: ':='-style assignments are not portable
> Makefile.am:176:   'aminclude_static.am' included from here
> doc/Makefile.am: error: required file 'build-aux/depcomp' not found
> doc/Makefile.am:   'automake --add-missing' can install 'depcomp'
> guile/Makefile.am:68: warning: AM_V_GUILEC_$(V: non-POSIX recursive variable expansion
> guile/Makefile.am:69: warning: AM_V_GUILEC_$(AM_DEFAULT_VERBOSITY: non-POSIX recursive variable expansion
> guile/Makefile.am:83: warning: '%'-style pattern rules are a GNU make extension
> guile/src/Makefile.am:117: warning: '%'-style pattern rules are a GNU make extension
> make[4]: *** [Makefile:2188: Makefile.in] Error 1
> make[4]: Leaving directory '/run/media/james/D/test_e8450/master/build_dir/target-aarch64_cortex-a53_musl/gnutls-3.7.8'
> make[3]: *** [Makefile:269: /run/media/james/D/test_e8450/master/build_dir/target-aarch64_cortex-a53_musl/gnutls-3.7.8/.built] Error 2
> make[3]: Leaving directory '/run/media/james/D/test_e8450/master/feeds/packages/libs/gnutls'
> time: package/feeds/packages/gnutls/compile#6.22#0.26#6.40
>     ERROR: package/feeds/packages/gnutls failed to build.
> make[2]: *** [package/Makefile:116: package/feeds/packages/gnutls/compile] Error 1
> make[2]: Leaving directory '/run/media/james/D/test_e8450/master'
> make[1]: *** [package/Makefile:110: /run/media/james/D/test_e8450/master/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
> make[1]: Leaving directory '/run/media/james/D/test_e8450/master'
> make: *** [/run/media/james/D/test_e8450/master/include/toplevel.mk:231: world] Error 2
> [james@ninja master]$ 
> `
> ```

i have this same issue related to a automake commit yesterday that broke gnutls but also kmod- but it unrelated to this commit. Please just compile a bare minimum system with no packages

PerkelSimon commented:
> i have this same issue related to a automake commit made yesterday that broke gnutls but also kmod- but it is unrelated to this commit. Please just compile a bare minimum system with no packages

Compile without  package, it is not what I need ;-)

rmandrad commented:
the issue with gnutls is now fixed ... just pull the latest from master

graysky2 commented:
@robimarko @mj22226 @hauke - I am having trouble rebasing 5.15.103, specifically these two patches:
```
pending-5.15/733-02-net-ethernet-mtk_eth_soc-fix-RX-data-corruption-issu.patch
backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch
```

Can one you guys take a look?  The first one seems straight forward (deleting the nearly matching code) but the second one, although only a single line addition, seemingly conflicts with something included in upstream's code.

https://github.com/graysky2/openwrt/commit/0f624b2b7ad7009b4a2825415018a227c3536e4d

robimarko commented:
Second one is conflicting with the following commit being backported:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/net/ethernet/mediatek?h=v5.15.103&id=2adc29350a5b4669544566f71f208d2abaec60ab

As it changed the `mcr_new`, basically it needs to be manually updated

graysky2 commented:
@robimarko - OK... I fixed those two but am finding that `pending-5.15/733-02-net-ethernet-mtk_eth_soc-fix-RX-data-corruption-issu.patch` wants to add a single line to `drivers/net/ethernet/mediatek/mtk_eth_soc.h` which is already defined....

```
%cat drivers/net/ethernet/mediatek/mtk_eth_soc.h.rej

--- drivers/net/ethernet/mediatek/mtk_eth_soc.h
+++ drivers/net/ethernet/mediatek/mtk_eth_soc.h
@@ -392,6 +392,7 @@
 #define MAC_MCR_FORCE_MODE     BIT(15)
 #define MAC_MCR_TX_EN          BIT(14)
 #define MAC_MCR_RX_EN          BIT(13)
+#define MAC_MCR_BIT_12         BIT(12)
 #define MAC_MCR_BACKOFF_EN     BIT(9)
 #define MAC_MCR_BACKPR_EN      BIT(8)
 #define MAC_MCR_FORCE_RX_FC    BIT(5)
```

And here is the relevant bit of `drivers/net/ethernet/mediatek/mtk_eth_soc.h`:
```
...
/* Mac control registers */
#define MTK_MAC_MCR(x)          (0x10100 + (x * 0x100))
#define MAC_MCR_MAX_RX_MASK     GENMASK(25, 24)
#define MAC_MCR_MAX_RX(_x)      (MAC_MCR_MAX_RX_MASK & ((_x) << 24))
#define MAC_MCR_MAX_RX_1518     0x0
#define MAC_MCR_MAX_RX_1536     0x1
#define MAC_MCR_MAX_RX_1552     0x2
#define MAC_MCR_MAX_RX_2048     0x3
#define MAC_MCR_IPG_CFG         (BIT(18) | BIT(16))
#define MAC_MCR_FORCE_MODE      BIT(15)
#define MAC_MCR_TX_EN           BIT(14)
#define MAC_MCR_RX_EN           BIT(13)
#define MAC_MCR_RX_FIFO_CLR_DIS BIT(12)
#define MAC_MCR_BACKOFF_EN      BIT(9)
#define MAC_MCR_BACKPR_EN       BIT(8)
#define MAC_MCR_FORCE_RX_FC     BIT(5)
#define MAC_MCR_FORCE_TX_FC     BIT(4)
#define MAC_MCR_SPEED_1000      BIT(3)
#define MAC_MCR_SPEED_100       BIT(2)
#define MAC_MCR_FORCE_DPX       BIT(1)
#define MAC_MCR_FORCE_LINK      BIT(0)
...
```

What is the right course of action?

robimarko commented:
That patch is the one I linked as backported in 5.15.103:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/net/ethernet/mediatek?h=v5.15.103&id=2adc29350a5b4669544566f71f208d2abaec60ab

So it should just be dropped then

graysky2 commented:
Damn, I need more sleep.  You are right, thanks for the tip.

graysky2 commented:
@robimarko - I still can't get this rebase right... I complete it here: https://github.com/graysky2/openwrt/commit/1664c1c404ebb904dd2f21fafb084bc5470aac97

But building fails for 
```
CONFIG_TARGET_ramips=y
CONFIG_TARGET_ramips_mt7621=y
CONFIG_TARGET_ramips_mt7621_DEVICE_tplink_archer-a6-v3=y
```

Here is a failure:
```
...
make[6]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[7]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[8]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[9]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
  CC      drivers/net/ethernet/mediatek/mtk_eth_soc.o
drivers/net/ethernet/mediatek/mtk_eth_soc.c: In function 'mtk_mac_config':
drivers/net/ethernet/mediatek/mtk_eth_soc.c:503:9: error: 'mcr_cur' undeclared (first use in this function)
  503 |         mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
      |         ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c:503:9: note: each undeclared identifier is reported only once for each function it appears in
drivers/net/ethernet/mediatek/mtk_eth_soc.c:504:9: error: 'mcr_new' undeclared (first use in this function)
  504 |         mcr_new = mcr_cur;
      |         ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c: In function 'mtk_mac_finish':
drivers/net/ethernet/mediatek/mtk_eth_soc.c:531:22: warning: unused variable 'mcr_new' [-Wunused-variable]
  531 |         u32 mcr_cur, mcr_new;
      |                      ^~~~~~~
drivers/net/ethernet/mediatek/mtk_eth_soc.c:531:13: warning: unused variable 'mcr_cur' [-Wunused-variable]
  531 |         u32 mcr_cur, mcr_new;
      |             ^~~~~~~
make[9]: *** [scripts/Makefile.build:289: drivers/net/ethernet/mediatek/mtk_eth_soc.o] Error 1
make[9]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[8]: *** [scripts/Makefile.build:552: drivers/net/ethernet/mediatek] Error 2
make[8]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[7]: *** [scripts/Makefile.build:552: drivers/net/ethernet] Error 2
make[7]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[6]: *** [scripts/Makefile.build:552: drivers/net] Error 2
make[6]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[5]: *** [Makefile:1912: drivers] Error 2
make[5]: Leaving directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103'
make[4]: *** [Makefile:23: /scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.103/.modules] Error 2
make[4]: Leaving directory '/scratch/union/target/linux/ramips'
make[3]: *** [Makefile:11: compile] Error 2
make[3]: Leaving directory '/scratch/union/target/linux'
time: target/linux/compile#2.52#0.71#3.13
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:30: target/linux/compile] Error 1
make[2]: Leaving directory '/scratch/union'
make[1]: *** [target/Makefile:23: /scratch/union/staging_dir/target-mipsel_24kc_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:231: world] Error 2
```

hauke commented:
I merged the update till kernel 5.15.102 now then it will get wider testing and so on.
